### PR TITLE
Fix overflow

### LINKED ISSUE #13: Fix overflow
Image

### PR DESCRIPTION
**Pull request by Suryakantdsa**
fix: resolve overflow issue in sidebar issue #13 

Screenshot (29)

[FIxed #13](https://github.com/code100x/mobile-magic/issues/13)

### CODE REVIEW

**Comment by Suryakantdsa** [comment]
## Now resolved the merge conflict @hkirat sir

This PR addresses the following:
1. **Overflow Issue**: Fixes the overflow issue in the sidebar 
2. **Authentication**: Wraps the sidebar content in the `<SignedIn>` component to ensure only authenticated users can view the project history.
3. **Hookify Project Fetching**: Replaces direct project fetching with the `useProjects` hook for better code organization and reusability.

Fixes #13

File: mobile-magic/apps/frontend/hooks/useProject.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+      const response = await axios.get(`${BACKEND_URL}/projects`, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      const projectsByDate = response.data.projects.reduce(
+        (acc: { [date: string]: Project[] }, project: Project) => {
+          const date = formatDate(project.createdAt);
+          //   const date = new Date(project.createdAt).toLocaleDateString("en-US", {
```
Comment (hkirat) [inline mobile-magic/apps/frontend/hooks/useProject.ts:27]: ?

Comment (Suryakantdsa) [reply mobile-magic/apps/frontend/hooks/useProject.ts:27]: @hkirat  sir this formatDate utils function takes a date string as input and returns a formatted string based on how recent the date is:
- If the date is today, it show in sidebar "Today".
- If the date is yesterday, it  show in sidebar "Yesterday".
- If the date is within the last 7 days, it  show in sidebar  "Last 7 Days".
- If the date is within the last 30 days, it  show in sidebar "Last Month".
- For dates older than that, it returns the date in a readable format (e.g., "January 1, 2023").

**Comment by Suryakantdsa** [comment]
Hi @hkirat  sir , just following up on my review comment regarding the formatDate utility function. Let me know if you have a chance to check it or if any further clarifications are needed. Thanks!🙏

File: mobile-magic/apps/frontend/hooks/useProject.ts (hunk 1 above)
Comment (Suryakantdsa) [reply mobile-magic/apps/frontend/hooks/useProject.ts]: @yashmakhija could you check this PR

File: mobile-magic/apps/frontend/hooks/useProject.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            acc[date] = [];
+          }
+          acc[date].push(project);
+          return acc;
+        },
+        {}
+      );
+      setProjects(projectsByDate);
+    })();
+  }, []);
```
Comment (yashmakhija) [inline mobile-magic/apps/frontend/hooks/useProject.ts:37]: Consider adding getToken to the dependency array to ensure the latest authentication token is used.

File: mobile-magic/apps/frontend/hooks/useProject.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  id: string;
+  description: string;
+  createdAt: string;
+};
+
+export function useProjects() {
+  const { getToken } = useAuth();
+  const [projects, setProjects] = useState<{ [date: string]: Project[] }>({});
+  useEffect(() => {
+    (async () => {
```
Comment (yashmakhija) [inline mobile-magic/apps/frontend/hooks/useProject.ts:16]: Instead of immediately invoking an async function inside useEffect, define it separately and call it inside the effect.

Comment (yashmakhija) [reply mobile-magic/apps/frontend/hooks/useProject.ts:16]: Why? React’s strict mode in development might trigger double execution, leading to multiple API calls.

File: mobile-magic/apps/frontend/hooks/useProject.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+type Project = {
+  id: string;
+  description: string;
+  createdAt: string;
+};
+
+export function useProjects() {
+  const { getToken } = useAuth();
+  const [projects, setProjects] = useState<{ [date: string]: Project[] }>({});
+  useEffect(() => {
```
Comment (yashmakhija) [inline mobile-magic/apps/frontend/hooks/useProject.ts:15]: Consider using useCallback for fetchProjects() and calling it inside useEffect to prevent unnecessary re-fetching.

File: mobile-magic/apps/frontend/components/ProjectsDrawer.tsx [hunk 5]
```diff
@@ -1,131 +1,132 @@
...
-        window.addEventListener('mousemove', handleMouseMove);
+              }}
+              variant="ghost"
+              className="w-full bg-blue-500/20 text-blue-600 hover:text-blue-400"
+            >
+              <MessageSquareIcon /> Start new project
+            </Button>
+            <DrawerTitle className="text-sm">Your projects</DrawerTitle>
+            <div className="flex space-between border rounded-md pr-2 pl-1 shadow-[0_3px_10px_rgb(0,0,0,0.2)]">
+              <input
```
Comment (yashmakhija) [inline mobile-magic/apps/frontend/components/ProjectsDrawer.tsx:68]: Consider using debounce  or a custom hook to optimize search input performance.

Comment (yashmakhija) [reply mobile-magic/apps/frontend/components/ProjectsDrawer.tsx:68]: Why? This can slow down performance, especially if there are many projects.

File: mobile-magic/apps/frontend/components/ProjectsDrawer.tsx [hunk 6]
```diff
@@ -1,131 +1,132 @@
...
+      if (e.clientX < 40) {
+        setIsOpen(true);
+      }
+      if (e.clientX > WIDTH) {
+        setIsOpen(false);
+      }
+    };
 
-    }, [getToken])
+    window.addEventListener("mousemove", handleMouseMove);
```
Comment (yashmakhija) [inline mobile-magic/apps/frontend/components/ProjectsDrawer.tsx:39]: Consider adding a small delay before opening the drawer to prevent accidental triggers.

Comment (yashmakhija) [reply mobile-magic/apps/frontend/components/ProjectsDrawer.tsx:39]: Why? Users might accidentally trigger the drawer when moving their cursor near the edge.

File: mobile-magic/apps/frontend/utils/formatDate.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  // for yesterday
+  if (date.toDateString() === yesterday.toDateString()) {
+    return "Yesterday";
+  }
+
+  // for last 7 days
+  const last7Days = new Date(today);
+  last7Days.setDate(today.getDate() - 7);
+  if (date >= last7Days) {
```
Comment (yashmakhija) [inline mobile-magic/apps/frontend/utils/formatDate.ts:20]: Check if conditions order: First, check Today and Yesterday, then Last 7 Days, to avoid overriding specific dates.

Comment (yashmakhija) [reply mobile-magic/apps/frontend/utils/formatDate.ts:20]: Why? It can cause "Today" or "Yesterday" to be incorrectly categorized.

File: mobile-magic/apps/frontend/hooks/useProject.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    (async () => {
+      const token = await getToken();
+
+      const response = await axios.get(`${BACKEND_URL}/projects`, {
```
[the remark this answers is not in the document]
Comment (yashmakhija) [reply mobile-magic/apps/frontend/hooks/useProject.ts:19]: Why? The component won’t know if the API call failed, leaving users without proper messaging.

File: mobile-magic/apps/frontend/hooks/useProject.ts (hunk 2 above)
Comment (yashmakhija) [reply mobile-magic/apps/frontend/hooks/useProject.ts:37]: Why? If getToken() changes, the effect won’t re-run, leading to potential stale data issues.

**Comment by Suryakantdsa** [comment]
@yashmakhija Got it! I'll make the changes and update you soon.

File: mobile-magic/apps/frontend/utils/formatDate.ts (hunk 7 above)
Comment (Suryakantdsa) [reply mobile-magic/apps/frontend/utils/formatDate.ts:20]: The checks for 'Today' and 'Yesterday' are done first, followed by 'Last 7 Days' and 'Last Month'. based on the logic there is no overlap .
Let me know any specific edge cases to test

**Comment by Suryakantdsa** [comment]
I've addressed all the requested changes. Please review them and let me know if any further updates are needed. Thanks @yashmakhija 🚀
